akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/scaladsl/ClusterSharding.scala

```diff
@@ -417,6 +417,27 @@ object EntityTypeKey {
  */
 @DoNotInherit trait EntityRef[-M] extends RecipientRef[M] { this: InternalRecipientRef[M] =>
 
+  /**
+   * The identifier for the particular entity referenced by this EntityRef.
+   *
+   * {{{
+   * // given sharding, typeKey
+   * sharding.entityRefFor(typeKey, "someId").entityId == "someId"  // always true
+   * }}}
+   */
+  def entityId: String
+
+  /**
+   * The name of the EntityTypeKey associated with this EntityRef.
+   */
+  def entityName: String
```

Comment:
I'm not in love with `entityName`: `entityType` to my mind makes it seem too much like its referring to the type of messages which can be sent to the entity.

Maybe `entityCategory` is better?

Reply:
Any reason to not have a `entityTypeKey` and return that rather than to reach into it? `EntityTypeKey.name` is public API.